src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,136 @@
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        class PersonsComparator implements Comparator<Person> {
+
+            @Override
+            public int compare(Person left, Person right) {
+                return Integer.compare(left.getAge(), right.getAge());
+            }
+        }
+
+        Arrays.sort(persons, new PersonsComparator());
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person left, Person right) {
+                return Integer.compare(left.getAge(), right.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person left, Person right) {
+                int difference = left.getLastName().compareTo(right.getLastName());
+                return difference != 0 ? difference
+                    : left.getFirstName().compareTo(right.getFirstName());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+
+        Predicate<Person> isFirstAge30Checker = new Predicate<Person>() {
+            @Override
+            public boolean apply(Person person) {
+                return Integer.compare(person.getAge(), 30) == 0;
+            }
+        };
+
+        Optional<Person> personOptional = FluentIterable.from(persons)
+            .firstMatch(isFirstAge30Checker);
+
+        if (personOptional.isPresent()) {
```

Comment:
You can use ofNull() in FluentIterable to get the value if it's present, or null if not 
(see https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Optional.html#orNull())